From c14992bf96345f34805fe9bcb67562a38fe609e7 Mon Sep 17 00:00:00 2001
From: YangYonggang <b31664@freescale.com>
Date: Tue, 28 Feb 2012 19:58:27 +0800
Subject: [PATCH 244/635] ENGR00175634 Camera preview would hang after
 recording about 30 seconds.

The root cause, the encode lock and memory lock caused deed lock.
Remove the encode lock to avoid the dead lock.
The encode lock could be removed for in putBufferCount function clear all
of the more encoding flag.

Signed-off-by: YangYonggang <b31664@freescale.com>
---
 mx6/libcamera/CameraHal.cpp | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/mx6/libcamera/CameraHal.cpp b/mx6/libcamera/CameraHal.cpp
index 6a55cbc..56254bc 100755
--- a/mx6/libcamera/CameraHal.cpp
+++ b/mx6/libcamera/CameraHal.cpp
@@ -1881,14 +1881,11 @@ Pic_out:
         unsigned int buf_index = pBuf - &mCaptureBuffers[0];
 
         Mutex::Autolock _l(pBuf->mBufferLock);
-
-        mEncodeLock.lock();
         if(!mRecordRunning &&  (mVideoBufferUsing[buf_index] == 1)
                 && (pBuf->refCount == 2)) {
             pBuf->refCount --;
             mVideoBufferUsing[buf_index] = 0;
         }
-        mEncodeLock.unlock();
 
         pBuf->refCount --;
         if(pBuf->refCount == 0) {
@@ -2265,7 +2262,6 @@ Pic_out:
                     EncBuf = &mPPbuf[enc_index];
                 }
 
-                mEncodeLock.lock();
                 if ((mMsgEnabled & CAMERA_MSG_VIDEO_FRAME) && mRecordRunning) {
                     nsecs_t timeStamp = systemTime(SYSTEM_TIME_MONOTONIC);
                     if (bDerectInput == true) {
@@ -2280,10 +2276,8 @@ Pic_out:
                     getBufferCount(&mCaptureBuffers[enc_index]);
                     mVideoBufferUsing[enc_index] = 1;
                     mDataCbTimestamp(timeStamp, CAMERA_MSG_VIDEO_FRAME, mVideoMemory, enc_index, mCallbackCookie);
-                    mEncodeLock.unlock();
                     break;
                 }
-                mEncodeLock.unlock();
                 break;
 
             case CMESSAGE_TYPE_STOP:
-- 
1.8.0

